dss/storage/bundles.py

```diff
@@ -119,3 +120,41 @@ def _latest_version_from_object_names(object_names: typing.Iterator[str]) -> str
             version = current_version
 
     return version
+
+
+def list_available_uuids(replica: str = None,
+                         prefix: typing.Optional[str] = None,
+                         per_page: int = PerPageBounds.per_page_max,
+                         search_after: typing.Optional[str] = None,
+                         token: typing.Optional[str] = None):
+    """Return a list of available uuid's in a given prefix, removes tombstones"""
+    kwargs = dict(bucket=Replica[replica].bucket, prefix=prefix, k_page_max=per_page)
+    if search_after:
+        kwargs['start_after_key'] = search_after
+    if token:
+        kwargs['token'] = token
+
+    storage_handler = Config.get_blobstore_handle(Replica[replica])
+    prefix_iterator = storage_handler.list_v2(**kwargs)  # note dont wrap this in enumerate() it looses the token
+    keys = dict()  # type: dict
+    uuid_list = list()
+    total_keys = 0
+
+    for key, meta in prefix_iterator:
+        uuid, version = key.split('.', 1)
+        if not version.endswith(TOMBSTONE_SUFFIX):
+            search_after = key
+            keys.setdefault(uuid.strip(f'{BUNDLE_PREFIX}/'), []).append(version)
+            total_keys += 1
+        elif TOMBSTONE_SUFFIX == version:
+            if keys.get(uuid) is not None:
```

Comment:
This is a nitpick, but is it possible for `keys[uuid]` to be set to `None`? I am not familiar enough with the code to say. If not, then `uuid in keys` would be a clearer membership test.